packages/frontend/component/src/components/auth-components/password-input/index.tsx

```diff
@@ -1,104 +1,200 @@
+import { type PasswordLimitsFragment } from '@affine/graphql';
 import { useAFFiNEI18N } from '@affine/i18n/hooks';
 import { passwordStrength } from 'check-password-strength';
-import type { FC } from 'react';
-import { useCallback, useEffect, useState } from 'react';
+import { type FC, useEffect, useMemo } from 'react';
+import { useCallback, useState } from 'react';
+import { z, type ZodCustomIssue, ZodIssueCode } from 'zod';
 
 import type { InputProps } from '../../../ui/input';
 import { Input } from '../../../ui/input';
 import * as styles from '../share.css';
 import { ErrorIcon } from './error';
+import { statusWrapper } from './style.css';
 import { SuccessIcon } from './success';
 import { Tag } from './tag';
 
-export type Status = 'weak' | 'medium' | 'strong' | 'maximum';
+export type Status = 'weak' | 'medium' | 'strong' | 'minimum' | 'maximum';
 
 export const PasswordInput: FC<
   InputProps & {
+    passwordLimits: PasswordLimitsFragment;
     onPass: (password: string) => void;
     onPrevent: () => void;
   }
-> = ({ onPass, onPrevent, ...inputProps }) => {
+> = ({ passwordLimits, onPass, onPrevent, ...inputProps }) => {
   const t = useAFFiNEI18N();
 
   const [status, setStatus] = useState<Status | null>(null);
   const [confirmStatus, setConfirmStatus] = useState<
     'success' | 'error' | null
   >(null);
+  const [canSubmit, setCanSubmit] = useState(false);
 
   const [password, setPassWord] = useState('');
   const [confirmPassword, setConfirmPassword] = useState('');
+  const validationSchema = useMemo(() => {
+    const { minLength, maxLength } = passwordLimits;
+    return z.string().superRefine((val, ctx) => {
+      if (val.length < minLength) {
+        ctx.addIssue({
+          code: ZodIssueCode.custom,
+          message: t['com.affine.auth.set.password.message.minlength']({
+            min: String(minLength),
+          }),
+          params: {
+            status: 'minimum',
+          },
+        });
+      } else if (val.length > maxLength) {
+        ctx.addIssue({
+          code: ZodIssueCode.custom,
+          message: t['com.affine.auth.set.password.message.maxlength']({
+            max: String(maxLength),
+          }),
+          params: {
+            status: 'maximum',
+          },
+        });
+      }
```

Comment:
why not use min().max() directly? error message can process in view after, and we can use error code `too_small` and `too_big` to identify the error:
<img width="1071" alt="image" src="https://github.com/toeverything/AFFiNE/assets/25152247/c9d91cb3-0231-475b-93b5-453b4d279413">

Reply:
We want to pass the status parameter, so use `ZodCustomIssue`.

Reply:
where we use the message? i see this i18n call already in the view

Reply:
Good catch. 
Removed.